cmd/webhook.go

```diff
@@ -233,29 +236,48 @@ func createPatch(pod *corev1.Pod, sidecarConfigTemplate *Config, annotations map
 			sidecarConfig := tempSidecarConfig.(*Config)
-			bucketName := string(secretList.Items[sec].Data["S3_BUCKET"])
-			sidecarConfig.Containers[0].Name = filerName + "-" + bucketName + "-bucket-containers"
+
+			// Bucket might be a full path with shares, meaning with slashes (path1/path2)
+			bucketMount := string(secretList.Items[sec].Data["S3_BUCKET"])
+
+			// Take last path of share and limit length to 15 characters to manage length (max 63 for container names).
+			// The 15 characters limit was chosen arbitrarily and could be revised
+			bucketNamePaths := strings.Split(bucketMount, "/")
+			bucketName := bucketNamePaths[len(bucketNamePaths)-1]
+			if len(bucketName) > 15 {
+				bucketName = bucketName[0:15]
+			}
+
+			filerBucketName := filerName + "-" + bucketName
+			// Add number to prevent duplicate
+			if slices.Contains(filerBucketList, filerBucketName) {
+				filerBucketName = filerBucketName + "-" + strconv.Itoa(len(filerBucketList)+1)
+			}
+			filerBucketList = append(filerBucketList, filerBucketName)
+
+			sidecarConfig.Containers[0].Name = filerBucketName + "-bucket-container"
 			sidecarConfig.Containers[0].Args = []string{"-c", "/goofys --cheap --endpoint " + string(secretList.Items[sec].Data["S3_URL"]) +
 				" --http-timeout 1500s --dir-mode 0777 --file-mode 0777  --debug_fuse --debug_s3 -o allow_other -f " +
-				bucketName + "/ /tmp"}
+				bucketMount + "/ /tmp; echo sleeping...; sleep infinity"}
 
-			sidecarConfig.Containers[0].Env[0].Value = "fusermount3-proxy-" + filerName + "-" + bucketName + "-" + pod.Namespace + "/fuse-csi-ephemeral.sock"
+			sidecarConfig.Containers[0].Env[0].Value = "fusermount3-proxy-" + filerBucketName + "-" + pod.Namespace + "/fuse-csi-ephemeral.sock"
 
 			sidecarConfig.Containers[0].Env[1].Value = string(secretList.Items[sec].Data["S3_ACCESS"])
 			sidecarConfig.Containers[0].Env[2].Value = string(secretList.Items[sec].Data["S3_SECRET"])
 
-			sidecarConfig.Containers[0].VolumeMounts[0].Name = "fuse-fd-passing-" + filerName + "-" + bucketName + "-" + pod.Namespace
-			sidecarConfig.Containers[0].VolumeMounts[0].MountPath = "fusermount3-proxy-" + filerName + "-" + bucketName + "-" + pod.Namespace
+			volumeMountName := "fuse-fd-passing-" + filerBucketName + "-" + pod.Namespace
+			sidecarConfig.Containers[0].VolumeMounts[0].Name = volumeMountName
+			sidecarConfig.Containers[0].VolumeMounts[0].MountPath = "fusermount3-proxy-" + filerBucketName + "-" + pod.Namespace
 
-			sidecarConfig.Volumes[0].Name = ("fuse-fd-passing-" + filerName + "-" + bucketName + "-" + pod.Namespace)
-			sidecarConfig.Volumes[1].Name = ("fuse-csi-ephemeral-" + filerName + "-" + bucketName + "-" + pod.Namespace)
-			sidecarConfig.Volumes[1].CSI.VolumeAttributes["fdPassingEmptyDirName"] = "fuse-fd-passing-" + filerName + "-" + bucketName + "-" + pod.Namespace
+			sidecarConfig.Volumes[0].Name = volumeMountName
+			sidecarConfig.Volumes[1].Name = ("fuse-csi-ephemeral-" + filerBucketName + "-" + pod.Namespace)
```

Comment:
check vol name length restriction, is it reasonable vs our naming convention?

Reply:
I have't found out what convention volume names use, but k8s has 3 different conventions it follows https://kubernetes.io/docs/concepts/overview/working-with-objects/names/

So it's either a 253 character limit or 63. To be safe, we can try to follow the 63 character limit